app/models/classification.rb

```diff
@@ -529,7 +529,11 @@ def find_tag
 
   def save_tag
     tag_name = Classification.name2tag(name, parent_id, ns)
-    self.tag = Tag.in_region(region_id).find_or_create_by(:name => tag_name)
+    if tag_id.present? || tag.present?
```

Comment:
is it possible to have `tag_id` `nil` but `tag` present ?

Reply:
### Using a factory is sometimes strange

When you create a category (using a factory) you can create using `:tag` or `:tag_id`.
So I'm pretty sure it is possible for the `tag` to be populated but the `tag_id.nil?`

### Optimization

I pulled out the optimization PR, so we could get this merged quicker.

In the case that there is a `tag_id`, the `name` attribute has not been changed, and `tag` has not been fetched yet, this will not lookup the `tag`.
The next step is to not fetch the `tag` in the inner block as well (which is in the next PR)

When pulling these apart, sometimes there is bleeding between the PRs.